torch/onnx/utils.py

```diff
@@ -247,18 +247,22 @@ def _model_to_graph(model, args, f, verbose=False, training=False,
 
     graph = _optimize_graph(graph, operator_export_type)
 
+    input_and_param_names = [val.uniqueName() for val in graph.inputs()]
+    param_names = input_and_param_names[len(input_and_param_names) - len(params):]
+    params_dict = dict(zip(param_names, params))
+
     # NB: ONNX requires complete information about output types, which might be
     # erased by some optimizations, so we need to set it explicitly again.
     if torch_out is not None:
         output_tensors, _ = torch._C._jit_flatten(torch_out)
         for output, tensor in zip(graph.outputs(), output_tensors):
             output.inferTypeFrom(tensor)
 
-    _set_input_and_output_names(graph, input_names, output_names)
+    _set_input_and_output_names(graph, input_names, output_names, params_dict)
```

Comment:
if we move the creation of `params_dict` after this statement, then we don't need to update the params_dict in `_set_input_and_output_names` any more.

Reply:
Removed the update logic in `_set_input_and_output_names`.